src/library-authoring/create-unit/CreateUnitModal.tsx

```diff
@@ -0,0 +1,108 @@
+import { useLibraryContext } from '../common/context/LibraryContext';
+import messages from './messages';
+import { useCreateLibraryContainer } from '../data/apiHooks';
+import { ToastContext } from '../../generic/toast-context';
+
+const CreateUnitModal = () => {
+  const intl = useIntl();
+  const {
+    libraryId,
+    isCreateUnitModalOpen,
+    closeCreateUnitModal,
+  } = useLibraryContext();
+  const create = useCreateLibraryContainer(libraryId);
+  const { showToast } = React.useContext(ToastContext);
+
+  const handleCreate = React.useCallback((values) => {
+    create
+      .mutateAsync({
+        containerType: 'unit',
+        ...values,
+      })
+      .then(() => {
+        closeCreateUnitModal();
+        // TODO: Navigate to the new unit
+        // navigate(`/library/${libraryId}/units/${data.key}`);
+        showToast(intl.formatMessage(messages.createUnitSuccess));
+      })
+      .catch(() => {
+        showToast(intl.formatMessage(messages.createUnitError));
+      });
+  }, []);
+
+  return (
+    <ModalDialog
+      title={intl.formatMessage(messages.createUnitModalTitle)}
+      isOpen={isCreateUnitModalOpen}
+      onClose={closeCreateUnitModal}
+      hasCloseButton
+      isFullscreenOnMobile
+    >
+      <ModalDialog.Header>
+        <ModalDialog.Title>
+          {intl.formatMessage(messages.createUnitModalTitle)}
+        </ModalDialog.Title>
+      </ModalDialog.Header>
+
+      <Formik
+        initialValues={{
+          displayName: '',
+        }}
+        validationSchema={
+          Yup.object().shape({
+            displayName: Yup.string()
+              .required(intl.formatMessage(messages.createUnitModalNameInvalid)),
+          })
+        }
+        onSubmit={handleCreate}
+      >
+        {(formikProps) => (
+          <>
+            <ModalDialog.Body className="mw-sm">
+              <Form onSubmit={formikProps.handleSubmit}>
+                <FormikControl
+                  name="displayName"
+                  label={(
+                    <Form.Label className="font-weight-bold h3">
+                      {intl.formatMessage(messages.createUnitModalNameLabel)}
+                    </Form.Label>
+                  )}
+                  value={formikProps.values.displayName}
+                  placeholder={intl.formatMessage(messages.createUnitModalNamePlaceholder)}
+                  controlClasses="pb-2"
+                />
+              </Form>
+            </ModalDialog.Body>
+            <ModalDialog.Footer>
+              <ActionRow>
+                <ModalDialog.CloseButton variant="tertiary">
+                  {intl.formatMessage(messages.createUnitModalCancel)}
+                </ModalDialog.CloseButton>
+                <Button
+                  variant="primary"
+                  onClick={formikProps.submitForm}
+                  disabled={formikProps.isSubmitting || !formikProps.isValid || !formikProps.dirty}
+                >
```

Comment:
NIT: Can we use `LoadingButton` from `src/generic/loading-button/index.jsx` to display progress on click.

Reply:
Fixed: https://github.com/openedx/frontend-app-authoring/pull/1741/commits/9054aa1d63a0892024a90c9d6e1355fb1c7b39a1

I also migrated the `LoadingButton` to typescript here: https://github.com/openedx/frontend-app-authoring/pull/1741/commits/f0e0c4e28a072f6224a0f938d9250e31888a1224

Reply:
Nice